Sources/PaymentButtons/PayPalButton.swift

```diff
@@ -113,69 +113,65 @@ public final class PayPalButton: PaymentButton {
-        ) {
-            button = PayPalButton(
-                fundingSource: .payPal,
-                color: color.color,
-                edges: edges,
-                size: size,
-                insets: insets,
-                label: label?.label
-            )
-            self.action = action
-        }
-       
-        // MARK: - UIViewRepresentable methods
-        // TODO: Make unit test for UIVRepresentable methods: https://engineering.paypalcorp.com/jira/browse/DTNOR-623
+/// PayPalButton for SwiftUI
+public struct PayPalButtonView: UIViewRepresentable {
 
-        public func makeCoordinator() -> Coordinator {
-            Coordinator(action: action)
-        }
+    private var action: () -> Void = { }
+
+    private var button: PayPalButton
+
+    /// Initialize a PayPalButton
+    /// - Parameters:
+    ///   - insets: Edge insets of the button, defining the spacing of the button's edges relative to its content.
+    ///   - color: Color of the button. Default to gold if not provided.
+    ///   - edges: Edges of the button. Default to softEdges if not provided.
+    ///   - size: Size of the button. Default to standard if not provided.
+    ///   - label: Label displayed next to the button's logo. Default to no label.
+    public init(
+        insets: NSDirectionalEdgeInsets? = nil,
+        color: PayPalButton.Color = .gold,
+        edges: PaymentButtonEdges = .rounded,
+        size: PaymentButtonSize = .standard,
+        label: PayPalButton.Label? = nil,
+        _ action: @escaping () -> Void = { }
+    ) {
+        button = PayPalButton(
+            fundingSource: .payPal,
+            color: color.color,
+            edges: edges,
+            size: size,
+            insets: insets,
+            label: label?.label
+        )
+        self.action = action
+    }
 
-        public func makeUIView(context: Context) -> UIView {
-            let view = UIView()
+    // MARK: - UIViewRepresentable methods
+    // TODO: Make unit test for UIVRepresentable methods: https://engineering.paypalcorp.com/jira/browse/DTNOR-623
 
-            view.addSubview(button)
-            button.addTarget(context.coordinator, action: #selector(Coordinator.onAction(_:)), for: .touchUpInside)
+    public func makeCoordinator() -> Coordinator {
+        Coordinator(action: action)
+    }
 
-            return view
-        }
+    public func makeUIView(context: Context) -> PaymentButton {
 
-        public func updateUIView(_ uiView: UIView, context: Context) {
-            context.coordinator.action = action
-        }
+        let button = button
+        button.addTarget(context.coordinator, action: #selector(Coordinator.onAction(_:)), for: .touchUpInside)
+        return button
+    }
+
+    public func updateUIView(_ uiView: PaymentButton, context: Context) {
+        context.coordinator.action = action
     }
 }
 
 // MARK: PayPalButton Preview
 
-struct PayPalButtonView: View {
+struct PayPalButtonUIView: View {
```

Comment:
Do we need this view? I see that we are referencing `PayPalCreditButtonView` in demo app. Same question for the Credit and Pay Later buttons.